lib/signs/realtime.ex

```diff
@@ -196,6 +211,25 @@ defmodule Signs.Realtime do
     {:noreply, state}
   end
 
+  defp has_service_ended_for_source?(sign, source, current_time) do
+    SourceConfig.sign_stop_ids(source)
+    |> Enum.all?(&has_last_trip_departed_stop?(&1, sign, current_time))
```

Comment:
Ah that's a good call out, it would be just one stop id and not all. And good question about the multiple branches too, I'll ask the transit data team about that one.

Reply:
So about routes, I'm not sure if it's something we need/want to handle right now as GL last trips are not going to be supported until last trips are a feature in Glides. It does sound like once it's implemented though, that it will be a last trip for every route, which means that we would probably need to key recent departures on stop_id + route, instead of just stop_id. It's not much more effort to include the route right now though so we can also just go ahead and do it to prepare for the future.

The routes discussion brings up an interesting case though. Since the red line has branching, there will be two last trips in both directions, but there aren't GTFS route_ids for the two red line branches so somehow the code will need to know to wait for two last trips at RL trunk stops, and just one for RL branch stops..